datastore/providers/chroma_datastore.py

```diff
@@ -0,0 +1,205 @@
+                {
+                    "created_at": {
+                        "$lte": int(
+                            datetime.fromisoformat(query_filter.end_date).timestamp()
+                        )
+                    }
+                },
+            ]
+        elif query_filter.start_date:
+            output["created_at"] = {
+                "$gte": int(datetime.fromisoformat(query_filter.start_date).timestamp())
+            }
+        elif query_filter.end_date:
+            output["created_at"] = {
+                "$lte": int(datetime.fromisoformat(query_filter.end_date).timestamp())
+            }
+
+        return output
+
+    def _process_metadata_for_storage(self, metadata: DocumentChunkMetadata) -> Dict:
+        return {
+            "source": metadata.source.value,
+            "source_id": metadata.source_id,
+            "url": metadata.url,
+            "created_at": int(datetime.fromisoformat(metadata.created_at).timestamp()),
+            "author": metadata.author,
+            "document_id": metadata.document_id,
+        }
+
+    def _process_metadata_from_storage(self, metadata: Dict) -> DocumentChunkMetadata:
+        return DocumentChunkMetadata(
+            source=Source(metadata["source"]),
+            source_id=metadata["source_id"],
+            url=metadata["url"],
+            created_at=datetime.fromtimestamp(metadata["created_at"]).isoformat(),
+            author=metadata["author"],
+            document_id=metadata["document_id"],
+        )
+
+    async def _query(self, queries: List[QueryWithEmbedding]) -> List[QueryResult]:
+        """
+        Takes in a list of queries with embeddings and filters and returns a list of query results with matching document chunks and scores.
+        """
+        results = [
+            self.collection.query(
+                query_embeddings=[query.embedding],
+                include=["documents", "distances", "metadatas", "embeddings"],
+                n_results=min(query.top_k, self.collection.count()),
+                where=(
+                    self._where_from_query_filter(query.filter) if query.filter else {}
+                ),
+            )
+            for query in queries
+        ]
+
+        output = []
+        for query, result in zip(queries, results):
+            inner_results = []
+            (ids,) = result["ids"]
+            (embeddings,) = result["embeddings"]
+            (documents,) = result["documents"]
+            (metadatas,) = result["metadatas"]
+            (distances,) = result["distances"]
+            for id_, embedding, text, metadata, distance in zip(
+                ids, embeddings, documents, metadatas, distances
+            ):
+                inner_results.append(
+                    DocumentChunkWithScore(
+                        id=id_,
+                        text=text,
+                        metadata=self._process_metadata_from_storage(metadata),
+                        embedding=embedding,
```

Comment:
Could you comment this out for now? Returning the whole embedding makes the response too large for ChatGPT (results in ResponseTooLarge error because of size limits on the plugin responses)

I should make this clearer in the documentation, and also plan to add an include_embeddings option to the query endpoint for all the vector dbs because it seems like some people are using this as a general retrieval API and not just as a ChatGPT plugin

Reply:
o can you help